crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2163,67 +2164,50 @@ impl<'db> Type<'db> {
                     };
 
                     match self.try_call_dunder(db, "__bool__", &CallArguments::none()) {
-                        ref result @ (Ok(ref outcome)
-                        | Err(CallDunderError::PossiblyUnbound(ref outcome))) => {
```

Comment:
This has to be separated out into two match arms since the `Err` outcome is boxed and the `Ok` one is not.